players/base.py

```diff
@@ -46,6 +46,9 @@ def __eq__(self, other):
         else:
             return self.id == other.id
 
+    def __hash__(self):
```

Comment:
+1 As I recall, this means we can put it in a dict as the key now. Very nice.

Reply:
Yes. It was also needed to get some of the code to work lol.